src/python/gudhi/_simplex_tree.cc

```diff
@@ -0,0 +1,410 @@
+computed with these values.
+
+.. note::
+
+    Note that after calling this function, the filtration values are actually modified within the simplex tree.
+    The function :func:`extended_persistence` retrieves the original values.
+
+.. note::
+
+    Note that this code creates an extra vertex internally, so you should make sure that the simplex tree does not
+    contain a vertex with the largest possible value (i.e., 4294967295).
+
+This `notebook <https://github.com/GUDHI/TDA-tutorial/blob/master/Tuto-GUDHI-extended-persistence.ipynb>`_
+explains how to compute an extension of persistence called extended persistence.
+           )doc")
+      .def("_collapse_edges",
+           &gsti::collapse_edges,
+           nb::call_guard<nb::gil_scoped_release>())
+      .def("reset_filtration",
+           &gsti::reset_filtration,
+           nb::arg("filtration"),
+           nb::arg("min_dim") = 0,
+           nb::call_guard<nb::gil_scoped_release>(),
+           R"doc(
+This function resets the filtration value of all the simplices of dimension at least min_dim. Resets all the simplex
+tree when `min_dim = 0`.
+`reset_filtration` may break the filtration property with `min_dim > 0`, and it is the user's responsibility to make it
+a valid filtration (using a large enough `filt_value`, or calling `make_filtration_non_decreasing` afterwards for
+instance).
+
+:param filtration: New threshold value.
+:type filtration: float.
+:param min_dim: The minimal dimension. Default value is 0.
+:type min_dim: int.
+           )doc")
+      .def(nb::self == nb::self, R"doc(
+Equality operator in order to compare 2 SimplexTree data structures.
+
+:returns: `True` if the 2 complexes have the same simplices with the same filtration values, `False` otherwise.
+:rtype: bool
+           )doc")
+      .def("get_simplices", &gsti::get_simplex_python_iterator, R"doc(
+This function returns a generator with simplices and their given filtration values.
+
+:returns:  The simplices.
+:rtype:  generator with tuples(simplex, filtration)
+          )doc")
+      .def("get_filtration", &gsti::get_filtration_python_iterator, R"doc(
+This function returns a generator with simplices and their given
+filtration values sorted by increasing filtration values.
+
+:returns:  The simplices sorted by increasing filtration values.
+:rtype:  generator with tuples(simplex, filtration)
+          )doc")
+      .def("get_skeleton", &gsti::get_skeleton_python_iterator, nb::arg("dimension"), R"doc(
+This function returns a generator with the (simplices of the) skeleton of a maximum given dimension.
+
+:param dimension: The skeleton dimension value.
+:type dimension: int
+:returns:  The (simplices of the) skeleton of a maximum dimension.
+:rtype:  generator with tuples(simplex, filtration)
+          )doc")
+      .def("get_boundaries", &gsti::get_boundary_python_iterator, R"doc(
+This function returns a generator with the boundaries of a given N-simplex.
```

Comment:
It is a bit confusing that the doc shows
```pre
get_boundaries(self, arg: collections.abc.Sequence[int], /) → collections.abc.Iterator[tuple[list[int], float]]
```
(it automatically deduces "Iterator" from the C++ return type?) but the text says "generator" (there is a collections.abc.Generator). Not sure we should change anything, but worth a quick look.

Reply:
Should we say "This function returns ***an iterator***" instead of "a generator" then, if it isn't a generator?